src/core/messages.ts

```diff
@@ -288,23 +525,44 @@ export class DefaultMessages
    * @param clientId The client ID of the user.
    * @param logger An instance of the Logger.
    */
-  constructor(roomId: string, channelManager: ChannelManager, chatApi: ChatApi, clientId: string, logger: Logger) {
+  constructor(
+    roomId: string,
+    options: MessageOptions,
+    channelManager: ChannelManager,
+    chatApi: ChatApi,
+    clientId: string,
+    logger: Logger,
+  ) {
     super();
     this._roomId = roomId;
+    this._options = options;
 
     this._channel = this._makeChannel(roomId, channelManager);
 
     this._chatApi = chatApi;
     this._clientId = clientId;
     this._logger = logger;
     this._listenerSubscriptionPoints = new Map<MessageListener, Promise<{ fromSerial: string }>>();
+
+    this.reactions = new DefaultMessageReactions(this._logger, options, this._chatApi, this._roomId, this._channel);
   }
 
   /**
    * Creates the realtime channel for messages.
    */
   private _makeChannel(roomId: string, channelManager: ChannelManager): Ably.RealtimeChannel {
-    const channel = channelManager.get(messagesChannelName(roomId));
+    const channelName = messagesChannelName(roomId);
+
+    if (this._options.rawMessageReactions) {
```

Comment:
At this point it may be too late to do the merge, as `channels.get` may have already been called - we usually do this in `room.getChannelManager`